api/src/test/java/org/pmiops/workbench/workspaces/WorkspaceServiceTest.java

```diff
@@ -159,7 +227,66 @@ public void activeStatus() {
     EnumSet.allOf(WorkspaceActiveStatus.class)
         .forEach(
             status ->
-                assertThat(mockDbWorkspace("1", "1", status).getWorkspaceActiveStatusEnum())
+                assertThat(
+                        buildDbWorkspace(
+                                workspaceIdIncrementer.getAndIncrement(),
+                                "1",
+                                DEFAULT_WORKSPACE_NAMESPACE,
+                                status)
+                            .getWorkspaceActiveStatusEnum())
                     .isEqualTo(status));
   }
+
+  @Test
+  public void updateRecentWorkspaces() {
+    workspaces.forEach(
+        workspace -> {
+          // Need a new 'now' each time or else we won't have lastAccessDates that are different
+          // from each other
+          workspaceService.updateRecentWorkspaces(
+              workspace.getWorkspaceId(),
+              USER_ID,
+              Timestamp.from(NOW.minusSeconds(workspaces.size() - workspace.getWorkspaceId())));
+        });
+    List<UserRecentWorkspace> recentWorkspaces =
+        workspaceService.getRecentWorkspacesByUser(USER_ID);
+    assertThat(recentWorkspaces.size()).isEqualTo(WorkspaceServiceImpl.RECENT_WORKSPACE_COUNT);
+    assertThat(
+            recentWorkspaces.stream()
```

Comment:
I'd make variables for these two lists like `expected` and `actual` and just do the assertion at the end so it's all on one line. Easier to read with fewer indentations plus it's simpler to step through in the debugger.